ember-server/src/main/scala/org/http4s/ember/server/internal/WebSocketHelpers.scala

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright 2019 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.ember.server.internal
```

Comment:
If there's nothing ember here, a more general internal package might be good.  There's renewed talk of servlet support, and it would look a lot like this with servlet's `HttpUpgradeHandler`.

Although if it's all internal, we can refactor it later.  Meh.  It's fine where it is.

Reply:
There isn't much ember-specific but it does rely on fs2's `Socket`, which we should be able to abstract over with some effort. I've been wanting to do that anyway so we can write some tests with more depth